receiver/awss3eventreceiver/internal/worker/worker.go

```diff
@@ -63,6 +64,34 @@ func parseS3Event(messageBody string) (*events.S3Event, error) {
 	return notification, err
 }
 
+// isDLQConditionError checks if an error should trigger DLQ behavior
+func isDLQConditionError(err error) bool {
```

Comment:
If this function returned the error type, the switch in ProcessMessage would not need to call each "is error" function a second time.

```go
type dlqErrType string

const (
  dlqErrAccessDenied = "isAccessDeniedError"
  dlqErrNoSuchKey = "isNoSuchKeyError"
  ...
)

func isDLQConditionError(err error) (bool, dlqErrType) {
```